src/Core/Repositories/TableStorage/EventRepository.cs

```diff
@@ -91,7 +93,7 @@ public async Task CreateManyAsync(IEnumerable<IEvent> e)
             return;
         }
 
-        var entities = e.Where(ev => ev is EventTableEntity).Select(ev => ev as EventTableEntity);
+        var entities = e.OfType<EventTableEntity>();
```

Comment:
`OfType` does the [same thing](https://github.com/dotnet/runtime/blob/ed53f555311c1cff94153c18aceb3f5997904b8e/src/libraries/System.Linq/src/System/Linq/OfType.cs#L50) we were already doing except by only iterating through the list once and maintaining null analysis.